include/buffer_tree.h

```diff
@@ -36,37 +38,45 @@ class BufferTree {
   // branching factor
   uint32_t B;
 
+  // number of nodes in the graph
+  Node N;
+
   // metadata control block(s)
-  std::vector<BufferControlBlock> buffers;
+  // level 1 blocks take indices 0->(B-1). So on and so forth from there
```

Comment:
so the root buffer is not in this vector?

Reply:
No, it could be but the rules for handling the root_node are a little different so I figured it would be easier to separate them out.